pkg/sql/sem/builtins/builtins.go

```diff
@@ -3597,6 +3598,49 @@ may increase either contention or retry errors, or both.`,
 		},
 	),
 
+	"crdb_internal.payloads_for_span": makeBuiltin(
+		tree.FunctionProperties{Category: categorySystemInfo},
+		tree.Overload{
+			Types:      tree.ArgTypes{{"span ID", types.Int}},
+			ReturnType: tree.FixedReturnType(types.Jsonb),
+			Fn: func(ctx *tree.EvalContext, args tree.Datums) (tree.Datum, error) {
+				builder := json.NewArrayBuilder(len(args))
+				var payloads []string
+				spanID := uint64(*(args[0].(*tree.DInt)))
+
+				var payload json.JSON
+				var err error
+				// Search all spans of the current trace for the span that matches the builtin's argument.
+				ctx.Settings.Tracer.VisitSpans(func(span *tracing.Span) error {
+					// TODO(angelapwen): Once a trace's activeSpans map is keyed on span ID, we can skip the iteration.
+					for _, rec := range span.GetRecording() {
+						if spanID == rec.SpanID {
+							rec.Structured(func(item *pbtypes.Any) {
+								payload, err = protoreflect.MessageToJSON(item, true /* emitDefaults */)
+								if err != nil {
+									return
+								}
+								if payload != nil {
+									builder.Add(payload)
+									payloads = append(payloads, payload.String())
```

Comment:
We don't need to be collecting to `payloads`, do we?

Reply:
Yep, you're right, this was a vestige of the string version we had played around with! Removing.